olp-cpp-sdk-core/src/cache/DefaultCacheImpl.cpp

```diff
@@ -259,17 +305,67 @@ bool DefaultCacheImpl::RemoveKeysWithPrefix(const std::string& key) {
     memory_cache_->RemoveKeysWithPrefix(key);
   }
 
+  if (mutable_cache_lru_) {
+    RemoveKeysWithPrefixLru(key);
+  }
+
   if (mutable_cache_) {
     return mutable_cache_->RemoveKeysWithPrefix(key);
   }
   return true;
 }
 
+void DefaultCacheImpl::InitializeLru() {
+  if (!mutable_cache_ || settings_.max_disk_storage == std::uint64_t(-1) ||
+      settings_.eviction_policy == EvictionPolicy::kNone) {
+    return;
+  }
+
+  mutable_cache_lru_ =
+      std::make_unique<DiskLruCache>(settings_.max_disk_storage);
+  auto it = mutable_cache_->NewIterator(leveldb::ReadOptions());
+  for (it->SeekToFirst(); it->Valid(); it->Next()) {
+    const auto& key = it->key();
+    const auto& value = it->value();
+
+    // Skip expiry keys
+    if (IsExpiryKey(key.ToString())) {
+      continue;
+    }
+
+    mutable_cache_lru_->Insert(key.ToString(), value.size());
+  }
+}
+
+void DefaultCacheImpl::RemoveKeysWithPrefixLru(const std::string& key) {
+  for (auto it = mutable_cache_lru_->begin();
```

Comment:
In `PromoteKeyLru()` you check if the `mutable_cache_lru` is null, here not. Why?